cmd/vulcan-host-discovery/main.go

```diff
@@ -140,13 +142,17 @@ nmapReportLoop:
 			"IP Address": strings.Join(hostIPAddresses, ","),
 			"Hostname":   strings.Join(hostHostnames, ","),
 		}
-		gr.Rows = append(gr.Rows, networkResource)
-	}
-
-	if add {
-		exposedVuln.Resources = append(exposedVuln.Resources, gr)
-		return []report.Vulnerability{exposedVuln}
+		gr.Rows = []map[string]string{networkResource}
+
+		if add {
+			exposedVuln.AffectedResource = networkResource["IP Address"]
+			exposedVuln.Resources = []report.ResourcesGroup{gr}
+			exposedVuln.Fingerprint = helpers.ComputeFingerprint(networkResource["IP Address"], networkResource["Hostname"])
```

Comment:
The IP is already part of the affected resource, so if it changes we'll get a new different finding. The fingerprint is meant to reflect the context of the current finding, so we shouldn't include nor the target nor content present in the affected resource field. The `Hostname` or nothing are good candidates for me.